hubblestack/log.py

```diff
@@ -154,10 +158,17 @@ def setup_splunk_logger():
     SPLUNK_HANDLER = handler
 
 
-def emit_to_splunk(message, level, name):
+def emit_to_splunk(message, level, name, remove_sensitive_logs=False):
     '''
     Emit a single message to splunk
     '''
+
+    if remove_sensitive_logs:
```

Comment:
filter_logs is trying to modify the "opts_to_log" object ( I might be wrong here ). I don't want it to edit the __grains__ so I am making a deepcopy here.

Though this is a computattionally expensive operation, I think we can tolerate this much expense since 'hubblestack.grains_report' is NOT generated very frequently.

@basepi  @jettero , please review my code from "memory leak" perspecitive also because __grain__ might be a huge object in some cases.

Reply:
I don't think this is necessary. `_remove_sensitive_info` uses list/dictionary comprehensions, which construct new objects. So by the time we get to the `remove_dots` modifications in `filter_logs` we are already in a copy.

@jettero would you mind confirming this for me? I want a second set of eyes.

Reply:
That seems right, except that if the obj passed to _remove_sensitive_info isn't a list or a dict (eg, a tuple, or a namedtuple) then it wouldn't be filtered or copied. Probably this isn't normally going to be the case, but I wonder if that section (isinstance(list) or isinstance(dict)) should have an else: raise ValueError("trying to filter !list/!dict, doh") or something like that.

Reply:
Except that we need to be able to emit things that aren't lists or dictionaries.

I recommend that we remove this deepcopy, but surround the `filter_logs()` call below with an `if isinstance(message, (list, dict)):` to make sure we only filter on lists/dicts.

Thanks @yassingh!

Reply:
Thanks for suggestions.
It's done 👍